### PR TITLE
Add Nightcore connector

### LINKED ISSUE #1355: Connector Request: Nightcore.com
After searching around I can't find any scrobbler that supports this site.
https://www.nightcore.com

Been using this addon for a long time so figured I would come  here to see if it is possible to add this site to the list of supported sites.

Not sure if it is possible since it uses embedded Youtube videos.

### PR DESCRIPTION
**Pull request by f1u77y**
Resolves #1355

### CODE REVIEW

**Comment by alexesprit** [comment]
I'm not sure the Youtube connector should be used for Nightcore. It always uses "Nightcore" as an artist name, whereas the artist name is in another element.

**Comment by f1u77y** [comment]
So I think we should take artist and title from the website but leave everything else to the YT connector, shouldn't we?

(And I couldn't find any other info on the website)

File: connectors/nightcore.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			return;
+		}
+		artist = event.data.artist;
+		track = event.data.track;
+	});
+
+	Connector.filter = new MetadataFilter({
+		artist: text => text.replace(/\s*original artist:/i, ''),
+		track: text => text.replace(/^nightcore - /i, ''),
+	});
```
Comment (alexesprit) [inline connectors/nightcore.js]: The filter doesn't trim song fields, but it should. You can use default `TrimFilter` with appended custom functions (see `MetadataFilter.append` function).

**Comment by inverse** [comment]
@f1u77y any chance for an update on this?

**Comment by alexesprit** [comment]
Closed due to inactivity.

**Pull request closed without merging**